packages/amplify_datastore/lib/types/DataStoreHubEvents/HubEventElementWithMetadata.dart

```diff
@@ -13,23 +13,56 @@
  * permissions and limitations under the License.
  */
 
-import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
-import 'HubEventElement.dart';
+part of 'HubEventElement.dart';
 
+/// The model and metadata associated with a DataStore `outboxMutationProcessed`
+/// Hub event.
 class HubEventElementWithMetadata extends HubEventElement {
-  late int version;
-  late int lastChangedAt;
-  late bool deleted;
-
-  HubEventElementWithMetadata(
-      Map<dynamic, dynamic> serializedData, ModelProviderInterface provider)
-      : super(serializedData, provider) {
-    Map<String, dynamic> serializedElement =
-        new Map<String, dynamic>.from(serializedData["element"]);
-    Map<String, dynamic> metadata =
-        new Map<String, dynamic>.from(serializedElement["syncMetadata"]);
-    version = metadata["_version"];
-    lastChangedAt = metadata["_lastChangedAt"];
-    deleted = metadata["_deleted"];
+  /// The version of the model.
+  final int version;
+
+  /// The last time the model was updated locally.
+  final DateTime lastChangedAt;
```

Comment:
I think we agreed to try to keep the Hub event as consistent as possible with other platforms; in this case, Android and iOS express lastChangedAt as an Int (I am not sure why). I would suggest leaving this as is (and if we do end up keeping this change we need to remember to list it as a breaking change).

Reply:
Yeah that was my bad, I forgot about breaking changes. Although chances are, if anyone is using that value, they are converting it to a datetime object first. In any case, I've reverted the change.